packages/flutter/lib/src/widgets/raw_menu_anchor.dart

```diff
@@ -173,19 +198,65 @@ class RawMenuAnchor extends StatefulWidget {
     this.consumeOutsideTaps = false,
     this.onOpen,
     this.onClose,
+    this.onOpenRequested,
+    this.onCloseRequested,
     this.useRootOverlay = false,
     this.builder,
     required this.controller,
     required this.overlayBuilder,
     this.child,
   });
 
-  /// A callback that is invoked when the menu is opened.
+  /// Called when the menu overlay is shown
+  ///
+  /// This callback is called when the menu overlay is added to the widget tree,
+  /// typically before opening animations begin. [onOpen] can be used to respond
+  /// when the menu first becomes interactive, such as by setting focus to the
+  /// menu.
+  ///
+  /// An open menu that is repositioned will not trigger [onOpen].
   final VoidCallback? onOpen;
 
-  /// A callback that is invoked when the menu is closed.
+  /// Called when the menu overlay is hidden.
+  ///
+  /// This callback is triggered after the menu overlay has been removed from
+  /// the widget tree, typically after all closing animations have completed. It
+  /// is typically used by applications to respond when the menu has been
+  /// dismissed.
```

Comment:
Slight fix, I wonder if this is the correct understanding.
```suggestion
  /// Called when the menu overlay is hidden.
  ///
  /// This callback is triggered when the menu overlay is removed from the widget
  /// tree, typically after any closing animations have completed. It is typically
  /// used by applications to respond when the menu has been dismissed, such as
  /// by restoring focus to a previously active element.
```

Also did you remove 
```
  /// It is not called when the menu is unmounted.
  /// By default it does nothing.
```
intentionally? I think the two sentences are worth adding. (The 2nd sentence also to `onOpen`).

Reply:
So, this is another confusing aspect of the menu. If the focus moves onto the menu, the menu takes focus when it begins opening (when onOpen is called), and moves off the menu when it begins closing (when onCloseRequested is called). I observed this in several menus. Otherwise, it'd be bad a11y -- the user would have to wait for the menu to open or close before that user could move to another focus node.

For the removal question, I'm not sure why I took out the first part ("It is not called when the menu is unmounted") -- it may have been an accident. The "by default it does nothing" part was removed since the value defaults to null, so I thought it was evident that the callback did nothing, but I do know there are classes that have default behavior if a null value is provided. So, I'll add them both back in.

Reply:
Do you mind if I rewrite it slightly to "By default, [onOpen] does nothing"? The rewording makes it a bit less ambigious since "menu" is also being referenced in the previous sentence.